UKHO.MaritimeSafetyInformation.Web/UKHO.MaritimeSafetyInformation.Common/Helpers/NMHelper.cs

```diff
@@ -125,5 +132,15 @@ public static async Task<byte[]> GetFileBytesFromStream(Stream stream)
             stream.Close();
             return fileBytes;
         }
+
+        public static List<ShowFilesResponseModel> ListFilesResponseCumulative(List<BatchDetails> batchDetails)
+        {
+            return GetShowFilesResponseModel(batchDetails)
+                .OrderByDescending(x => Convert.ToDateTime(x.Attributes.FirstOrDefault(y => y.Key == "BatchPublishedDate").Value))
+                .GroupBy(x => x.Attributes.FirstOrDefault(y => y.Key == "Data Date").Value)
+                .Select(grp => grp.First())
+                .OrderByDescending(x => Convert.ToDateTime(x.Attributes.FirstOrDefault(y => y.Key == "Data Date").Value))
```

Comment:
If these `.FirstOrDefault`s could return `null` you should use conditional access:
```suggestion
                .OrderByDescending(x => Convert.ToDateTime(x.Attributes.FirstOrDefault(y => y.Key == "BatchPublishedDate")?.Value))
                .GroupBy(x => x.Attributes.FirstOrDefault(y => y.Key == "Data Date")?.Value)
                .Select(grp => grp.First())
                .OrderByDescending(x => Convert.ToDateTime(x.Attributes.FirstOrDefault(y => y.Key == "Data Date")?.Value))
```

...or if not, change them to `.First`:
```suggestion
                .OrderByDescending(x => Convert.ToDateTime(x.Attributes.First(y => y.Key == "BatchPublishedDate").Value))
                .GroupBy(x => x.Attributes.First(y => y.Key == "Data Date").Value)
                .Select(grp => grp.First())
                .OrderByDescending(x => Convert.ToDateTime(x.Attributes.First(y => y.Key == "Data Date").Value))
```

Reply:
Changes are done as requested using ?